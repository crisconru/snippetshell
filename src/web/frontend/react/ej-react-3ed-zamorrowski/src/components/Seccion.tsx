import { useState } from 'react';

interface Props {
  texto?: string,
  fondo?: string,
  colorTexto?: string,
  children?: any
}

const CollapsableSeccion = ({texto, fondo, colorTexto, children}: Props) => {
  const [collapse, setCollapse] = useState(true)
  const toggleCollape = () => setCollapse(!collapse)
  return <>
    <h1 style={{backgroundColor: fondo, color: colorTexto}} onClick={toggleCollape}>{texto}</h1>
    <div style={{display: collapse ? 'none' : 'block'}}>
      {children}
    </div>
  </>
}

export const Clase = ({texto, children}: Props) => <CollapsableSeccion texto={texto} fondo='black' colorTexto='white' children={children}/>
export const Seccion = ({texto, children}: Props) => <CollapsableSeccion texto={texto} fondo='red' colorTexto='black' children={children}/>
export const Ejercicio = ({texto, children}: Props) => <CollapsableSeccion texto={texto} fondo='green' colorTexto='white' children={children}/>
