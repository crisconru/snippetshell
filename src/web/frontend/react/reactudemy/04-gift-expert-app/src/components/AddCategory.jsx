import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState('')

  const handleInputChange = (e) => {
    setInputValue(e.target.value)
    // console.log(`cambiada caja de texto con ${e.target.value}`)
  }

  const handleSubmit = (e) => {
    e.preventDefault()
    if (inputValue.trim().length > 0) {
      setCategories((categories) => [inputValue, ...categories])
      setInputValue('')
    }
    // console.log('submit hecho')
  }

  return (
    <form onSubmit={handleSubmit}>
      <h2>Add Category</h2>
      <input type='text' value={inputValue} onChange={handleInputChange} />
    </form>
  )
}

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired
}
