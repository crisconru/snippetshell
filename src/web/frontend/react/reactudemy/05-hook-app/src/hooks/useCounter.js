import { useState } from 'react';

const useCounter = (initialstate = 10) => {
  const [counter, setCounter] = useState(initialstate)

  const increment = (factor = 1) => {
    setCounter(counter + factor)
  }

  const decrement = (factor = 1) => {
    setCounter(counter - factor)
  }

  const reset = () => setCounter(initialstate)

  return {
    counter,
    increment,
    decrement,
    reset
  }
}

export default useCounter
