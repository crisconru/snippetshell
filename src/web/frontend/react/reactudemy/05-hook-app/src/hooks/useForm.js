import { useState } from "react";

const useForm = (initialstate = {}) => {
  const [values, setValues] = useState(initialstate);

  const reset = () => {
    setValues(initialstate)
  }

  const handleInputChange = ({ target }) => {
    setValues({
      ...values,
      [target.name]: target.value,
    });
  };
  // reset()
  return [values, handleInputChange, reset];
};

export default useForm;
