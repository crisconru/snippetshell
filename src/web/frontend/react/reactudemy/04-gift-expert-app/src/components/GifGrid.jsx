import React from 'react'
import PropTypes from 'prop-types'
import { GifGridItem } from './GifGridItem'
import { useFetchGifs } from '../hooks/useFetchGifs'

export const GifGrid = ({ category }) => {
  const { loading, data: images } = useFetchGifs(category)

  return (
    <>
      <h3>{category}</h3>
      {loading && (
        <p className='animate__animated animate_flash'>Cargando...</p>
      )}
      <div className='card-grid'>
        {images.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </div>
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}
