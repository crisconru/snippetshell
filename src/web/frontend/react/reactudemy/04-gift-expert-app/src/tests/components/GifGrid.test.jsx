import React from 'react'
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import { GifGrid } from '../../components/GifGrid'

import { useFetchGifs } from '../../hooks/useFetchGifs'

jest.mock('../../hooks/useFetchGifs')

describe('Pruebas para <GifGrid />', () => {
  const category = 'spiderman'
  let wrapper
  const gifs = [
    { id: 'ABC', url: 'http://localhost/algo.jpg', title: 'Cosa' },
    { id: 'DEF', url: 'http://localhost/algo.jpg', title: 'Cosa' }
  ]

  beforeEach(() => {
    jest.clearAllMocks()
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true
    })
    wrapper = shallow(<GifGrid category={category} />)
  })

  test('Debe mostrarse correctamente', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('Debe de mostrar items cuando se cargan imagenes useFetchgifs', () => {})
  useFetchGifs.mockReturnValue({
    data: gifs,
    loading: false
  })
  wrapper = shallow(<GifGrid category={category} />)
  // expect(wrapper).toMatchSnapshot()
  expect(wrapper.find('p').exists()).toBe(false)
  expect(wrapper.find('GifGridItem').length).toBe(gifs.length)
})
