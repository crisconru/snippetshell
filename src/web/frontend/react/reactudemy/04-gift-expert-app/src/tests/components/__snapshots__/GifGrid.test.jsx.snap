// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para <GifGrid /> Debe mostrarse correctamente 1`] = `
<Fragment>
  <h3>
    spiderman
  </h3>
  <p
    className="animate__animated animate_flash"
  >
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
